activation/post.go

```diff
@@ -46,6 +46,15 @@ type PostSetupOpts struct {
 	ComputeBatchSize  uint64              `mapstructure:"smeshing-opts-compute-batch-size"`
 }
 
+// InitializationConfig serves as a holder for config generated in
+// InitializerConfig to be passed to StartSession. It is not intended to be
+// user configurable.
+type InitializationConfig struct {
+	init            *initialization.Initializer
```

Comment:
What is the reasoning behind the config holding an instance of the `Initializer`?

Reply:
The `Initializer` constructor has many paths that return errors, so if we wan't to get the most possible validation up front, then we should call the constructor as part of validation. Once we have an instance of the `Initializer` it makes sense to use it, instead of duplicating the code to build it elsewhere.

Reply:
This is true, but calling the initiailzer twice with the same parameters shouldn't return different results in terms of validation.

The reason is that I aim to eventually retire the Initializer service for just an Initialize function. The Initialization service has no real state; the process just needs to be able to be cancelled. The function would be no-op (or return a know error like `ErrAlreadyInitialized`) when called and the initialization is already done.

Alternatively you can call this `PostSession` and remove the `PostSetupOpts`, they are afaik only needed to instantiate the Initialization Service.

Reply:
>The reason is that I aim to eventually retire the Initializer service for just an Initialize function. The Initialization service has no real state; the process just needs to be able to be cancelled. The function would be no-op (or return a known error like ErrAlreadyInitialized) when called and the initialization is already done.

How were you planning to return the errors from the function and feed them back to the user?

Regarding `PostSetupOpts` they are needed see - https://github.com/spacemeshos/go-spacemesh/pull/4319#discussion_r1175753970

`NewInitializer` opens files and saves files so it's not guaranteed to succeed if called again with the same params.